### PR TITLE
docker-compose 1.1.0

### PR DESCRIPTION
**Pull request by kriswill**
Fig renamed to "Docker Compose", version bump to 1.1.0

### CODE REVIEW

**Comment by hacfi** [comment]
:+1:

**Comment by tdsmith** [comment]
We don't support renames well so I think this will have to be done the other way, sorry, where `docker-compose` becomes an alias for `fig`.

**Comment by kriswill** [comment]
I guess we could wait until they change their website, since the tool is no longer called "Fig".  see: https://github.com/docker/compose/blob/master/CHANGES.md#110-2015-02-25  I aliased to fig as a convenience to allow old scripts to continue to work with the new binary, until people got used to the new name.

**Comment by tdsmith** [comment]
Okay, since nothing depends_on fig and the name of the executable is changing it makes sense to me to remove the old formula and add a new one; `brew upgrade fig` won't work but c'est la guerre. Can you remove fig.rb formula in this PR as well, if you're adding an alias?

File: Library/Formula/docker-compose.rb [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+  conflicts_with "fig", :because => "Branding change from Fig to Docker Compose."
+
+  bottle do
+    cellar :any
+    revision 1
+    sha1 "af332540bdc40e6e574d92ca7169c71071af9a5d" => :yosemite
+    sha1 "1f29c6947f4d6a797041dae82d0e5e2463224043" => :mavericks
+    sha1 "53a11121643db837d8d684859e9141643366f63c" => :mountain_lion
+  end
```
Comment (tdsmith) [inline Library/Formula/docker-compose.rb]: Please remove the bottle block; our bot will provide the right incantation when we pull.

File: Library/Formula/docker-compose.rb [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+  depends_on :python if MacOS.version <= :snow_leopard
+  depends_on "libyaml"
+  depends_on "docker" => :recommended
+  depends_on "boot2docker" => :recommended
```
Comment (tdsmith) [inline Library/Formula/docker-compose.rb]: What does it mean to install this package with or without support for docker or boot2docker? If I run `brew install docker-compose` does it make sense that Homebrew docker and boot2docker will be installed if they weren't installed previously? Is the goal to allow an exit for users who have installed docker some other way?

File: Library/Formula/docker-compose.rb [hunk 3]
```diff
@@ -0,0 +1,79 @@
+class DockerCompose < Formula
+  homepage "http://www.fig.sh/"
+  url "https://github.com/docker/compose/archive/1.1.0.tar.gz"
+  sha1 "175066934c19f455606b16f1b4e4b9f26fc3f599"
+
+  conflicts_with "fig", :because => "Branding change from Fig to Docker Compose."
```
Comment (tdsmith) [inline Library/Formula/docker-compose.rb]: I think this is unnecessary.

**Comment by tdsmith** [comment]
> I aliased to fig as a convenience to allow old scripts to continue to work with the new binary

To make sure we're on the same page, that alias affects `brew install`, not what happens later; you'd have to do something like `ln_s bin/"docker-compose", bin/"fig"` to have people get "docker-compose" when they run "fig". (And then it would conflict_with fig.)

**Comment by mikemcquaid** [comment]
> Okay, since nothing depends_on fig and the name of the executable is changing it makes sense to me to remove the old formula and add a new one; brew upgrade fig won't work but c'est la guerre. Can you remove fig.rb in this PR as well, if you're adding an alias?

I don't agree here, I think we should stick with the old name. I know it's annoying but it's better than silently breaking upgrades.

File: Library/Formula/docker-compose.rb [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+    ENV.prepend_create_path "PYTHONPATH", libexec/"lib/python2.7/site-packages"
+    system "python", *Language::Python.setup_install_args(libexec)
+
+    bin.install Dir[libexec/"bin/*"]
+    bin.env_script_all_files(libexec/"bin", :PYTHONPATH => ENV["PYTHONPATH"])
+  end
+
+  test do
+    system "#{bin}/docker-compose", "--version"
```
Comment (mikemcquaid) [inline Library/Formula/docker-compose.rb]: Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

**Comment by hacfi** [comment]
Someone who doesn’t know that `docker-compose` used to be `fig` won’t find it then. Or are you talking about an alias?

**Comment by mikemcquaid** [comment]
Yeh, create the alias in reverse so `fig` has a `docker-compose` alias. That's what we normally do for upgrades.

File: Library/Formula/docker-compose.rb (hunk 2 above)
Comment (kriswill) [inline Library/Formula/docker-compose.rb]: yes, it's possible to install Docker and boot2docker manually, that's why I just left it as recommended.

File: Library/Formula/docker-compose.rb (hunk 3 above)
Comment (kriswill) [inline Library/Formula/docker-compose.rb]: I will remove the `conflicts_with`

**Comment by kriswill** [comment]
https://github.com/docker/compose/issues/1024 back linking for reference here.

File: Library/Formula/docker-compose.rb (hunk 4 above)
Comment (kriswill) [inline Library/Formula/docker-compose.rb]: I'm not sure how I can write a better test, other than allowing fig to invoke docker, brew-cask, virtualbox and boot2docker.  Is installing a VM on all 3 mac minis on Jenkins really recommneded?  for now going with `docker-compose help build`, which is more complex than version, and demonstrates that the python script is parsing command line parameters correctly.

Comment (mikemcquaid) [inline Library/Formula/docker-compose.rb]: `help build` will work. Thanks!

**Comment by kriswill** [comment]
https://github.com/docker/docker/issues/9694

**Comment by kriswill** [comment]
```sh
→ brew info docker-compose
fig: stable 1.1.0
http://docs.docker.com/compose/
```
Is there any way we could make it say:
```sh
docker-compose: stable 1.1.0
```
instead?

**Comment by mikemcquaid** [comment]
Unfortunately not; if we do that we will break upgrades. We have other software in the same situation, unfortunately.

**Comment by kriswill** [comment]
@mikemcquaid Okay, are you satisfied with the tweaks I've made?  The only thing I couldn't do is a full-stack test.

File: Library/Formula/fig.rb [hunk 5]
```diff
@@ -68,9 +64,24 @@ def install
...
+      Each of these packages can be installed thru homebrew, or manually installed.
+
+        brew install boot2docker
+        brew install docker
+
+      Or, to manually install follow instructions here:
+
+        http://docs.docker.com/installation/mac/
+
+    EOS
```
Comment (mikemcquaid) [inline Library/Formula/fig.rb]: I think we can remove these caveats; they are already implied/specified by the dependencies above.

**Comment by LinusU** [comment]
Actually, just symlinking `fig` to `docker-compose` will still break all the scripts because this new version uses a file named `docker-compose.yml` instead of a file called `fig.yml`.

I think that the best thing might be to leave `fig.rb` and add a caveat that it's no longer supported and that you should migrate to `docker-compose`. Then add `docker-compose` as a new formula.

This way the user can choose to install both the tool and be able to work on projects using the old config `fig.yml` and projects with the new config `docker-compose.yml`.

*To clarify, `fig.yml` is not a config file for the program but rather the file that `fig` operates on, it can't do anything without it.*

**Comment by mikemcquaid** [comment]
Ok, sounds reasonable in this case (if a bit crappy for upstream to provide no automatic migration path)

**Comment by idyll** [comment]
#35272 should be reopened as well as it goes hand in hand with docker-compose and docker-swarm.

**Comment by kriswill** [comment]
@LinusU docker-compose will still operate correctly with existing fig.yml files as a fallback.

**Comment by kriswill** [comment]
@idyll docker-compose doesn't depend on docker-machine to operate, and I cannot find any reference to it in the docs for docker-compose.

**Comment by kriswill** [comment]
@mikemcquaid do you want me to revert it back to `docker-compose.rb` with alias to `fig`?  Also, what's going on with Jenkins?  It's taking forever for it to run the integration tests.

**Comment by mikemcquaid** [comment]
@BrewTestBot test this please

**Comment by mikemcquaid** [comment]
@kriswill We've been having issues with our CI infrastructure as a machine seems to be dying. @tdsmith, thoughts?

File: Library/Formula/fig.rb [hunk 6]
```diff
@@ -68,9 +64,10 @@ def install
 
     bin.install Dir[libexec/"bin/*"]
     bin.env_script_all_files(libexec/"bin", :PYTHONPATH => ENV["PYTHONPATH"])
+    ln_s bin/"docker-compose", bin/"fig"
   end
 
   test do
-    system "#{bin}/fig", "--version"
+    system "#{bin}/docker-compose", "help", "build"
```
Comment (tdsmith) [inline Library/Formula/fig.rb]: This gives an error exit code, unfortunately: http://bot.brew.sh/job/Homebrew%20Pull%20Requests/22020/version=mountain_lion/testReport/junit/brew-test-bot/mountain_lion/test_fig/

Comment (kriswill) [inline Library/Formula/fig.rb]: you're right it returns a `1`

```sh
kris git[docker-compose] /usr/local 
→ docker-compose help build
Build or rebuild services.

Services are built once and then tagged as `project_service`,
e.g. `composetest_db`. If you change a service's `Dockerfile` or the
contents of its build directory, you can run `compose build` to rebuild it.

Usage: build [options] [SERVICE...]

Options:
    --no-cache  Do not use cache when building the image.

kris git[docker-compose] /usr/local 
→ echo $?
1
```

**Comment by tdsmith** [comment]
I retract my earlier waffling; it's lousy to break upgrades and we shouldn't do it. Let's keep `fig.rb` until the long-awaited day, told of in prophecy, when Homebrew gets proper rename handling.

**Comment by kriswill** [comment]
@tdsmith I reverted the test, so it should pass the integration tests.  Let me know if there's anything else you would like.

**Comment by tdsmith** [comment]
Thanks for the pull request! 🎉 Homebrew depends on contributions from community members like you and we're grateful for your support & appreciate your patience with revisions.

**Pull request closed without merging**